### ISSUE TITLE
Dreame D10Plus

### ISSUE THREAD

**Issue by 53l3cu5** [rytilahti/python-miio#1714]
**Device information:**
  - Name(s) of the device: Dreame D10Plus
  - Link: https://www.amazon.fr/gp/product/B09TWBQH53

Use `miiocli device --ip <ip address> --token <token> info`.
  - Model: dreame.vacuum.r2205
  - Hardware version: Linux
  - Firmware version: 4.1.8_1078

**Additional context**
- In full charge and no activity:
	miiocli dreamevacuum --ip <ip address> --token <token> status

	WARNING:miio.device:Found an unsupported model 'dreame.vacuum.r2205' for class 'DreameVacuum'. If this is working for you, please open an issue at https://github.com/rytilahti/python-miio/
	WARNING:miio.miot_device:Unable to find mapping for dreame.vacuum.r2205, falling back to dreame.vacuum.mc1808
	ERROR:miio.integrations.vacuum.dreame.dreamevacuum_miot:Unknown ChargingStats (0)
	ERROR:miio.integrations.vacuum.dreame.dreamevacuum_miot:Unknown model for cleaning mode (dreame.vacuum.r2205)
	ERROR:miio.integrations.vacuum.dreame.dreamevacuum_miot:Unknown FaultStatus (100)
	ERROR:miio.integrations.vacuum.dreame.dreamevacuum_miot:Unknown OperatingMode (None)
	Battery level: 13
	Brush life level: None
	Brush left time: None
	Charging state: None
	Cleaning mode: None
	Device fault: None
	Device status: Sweeping
	Filter left level: None
	Filter life level: None
	Life brush main: None
	Life brush side: None
	Life sieve: None
	Map view: None
	Operating mode: None
	Side cleaning brush left time: None
	Side cleaning brush life level: None
	Time zone: None
	Timer enabled: None
	Timer start time: None
	Timer stop time: None
	Voice package: None
	Volume: None
	Water flow: None
	Water box attached: None 
	Cleaning time: None
	Cleaning area: None
	First clean time: None
	Total clean time: None
	Total clean times: None
	Total clean area: None

- At 92% of charge and in cleaning mode:
	miiocli dreamevacuum --ip <ip address> --token <token> status

	WARNING:miio.device:Found an unsupported model 'dreame.vacuum.r2205' for class 'DreameVacuum'. If this is working for you, please open an issue at https://github.com/rytilahti/python-miio/
	WARNING:miio.miot_device:Unable to find mapping for dreame.vacuum.r2205, falling back to dreame.vacuum.mc1808
	ERROR:miio.integrations.vacuum.dreame.dreamevacuum_miot:Unknown ChargingStats (0)
	ERROR:miio.integrations.vacuum.dreame.dreamevacuum_miot:Unknown model for cleaning mode (dreame.vacuum.r2205)
	ERROR:miio.integrations.vacuum.dreame.dreamevacuum_miot:Unknown FaultStatus (92)
	ERROR:miio.integrations.vacuum.dreame.dreamevacuum_miot:Unknown OperatingMode (None)
	Battery level: 1
	Brush life level: None
	Brush left time: None
	Charging state: None
	Cleaning mode: None
	Device fault: None
	Device status: Idle
	Filter left level: None
	Filter life level: None
	Life brush main: None
	Life brush side: None
	Life sieve: None
	Map view: None
	Operating mode: None
	Side cleaning brush left time: None
	Side cleaning brush life level: None
	Time zone: None
	Timer enabled: None
	Timer start time: None
	Timer stop time: None
	Voice package: None
	Volume: None
	Water flow: None
	Water box attached: None 
	Cleaning time: None
	Cleaning area: None
	First clean time: None
	Total clean time: None
	Total clean times: None
	Total clean area: None

**Comment by rytilahti** [comment]
Try `miiocli genericmiot` on the current git master: https://github.com/rytilahti/python-miio/#controlling-modern-miot-devices

**Comment by 53l3cu5** [comment]
I ended up not continuing to use
python-miio because my raspi 4B was too slow after installing crypto.
I couldn't go any further. I restored my system and I am looking for another less resource-intensive solution.
Thanks for your help

**Comment by TxMat** [comment]
> Try `miiocli genericmiot` on the current git master: https://github.com/rytilahti/python-miio/#controlling-modern-miot-devices

```sh
❯ miiocli genericmiot --ip 192.168.1.73 --token 357064496f434e4963666c5330736c7a status
Running command status
ERROR:miio.click_common:Exception: Cache file /home/uranus/.cache/python-miio/dreame.vacuum.r2205.json not found or it is stale
Traceback (most recent call last):
  File "/home/uranus/.local/lib/python3.11/site-packages/miio/click_common.py", line 54, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/uranus/.local/lib/python3.11/site-packages/miio/click_common.py", line 305, in wrap
    result = kwargs["result"] = func(*args, **kwargs)
                                ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/uranus/.local/lib/python3.11/site-packages/miio/click_common.py", line 270, in command_callback
    return miio_command.call(miio_device, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/uranus/.local/lib/python3.11/site-packages/miio/click_common.py", line 218, in call
    return method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/uranus/.local/lib/python3.11/site-packages/miio/click_common.py", line 184, in _wrap
    self._fetch_info()
  File "/home/uranus/.local/lib/python3.11/site-packages/miio/integrations/genericmiot/genericmiot.py", line 199, in _fetch_info
    self.initialize_model()
  File "/home/uranus/.local/lib/python3.11/site-packages/miio/integrations/genericmiot/genericmiot.py", line 65, in initialize_model
    self._miot_model = miotcloud.get_device_model(self.model)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/uranus/.local/lib/python3.11/site-packages/miio/miot_cloud.py", line 87, in get_device_model
    spec = self._file_from_cache(file)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/uranus/.local/lib/python3.11/site-packages/miio/miot_cloud.py", line 128, in _file_from_cache
    raise FileNotFoundError("Cache file %s not found or it is stale" % file)
FileNotFoundError: Cache file /home/uranus/.cache/python-miio/dreame.vacuum.r2205.json not found or it is stale

```

unfortunately it seems that miiocli needs a configuration file for the device to be able to run

**Comment by rytilahti** [comment]
@TxMat I just merged #1819 which should fix the cache issue, feel free to git pull and retry. If it's still not working, please open a new issue.

**Comment by TxMat** [comment]
working now !
thanks
